web/courses_new.go

```diff
@@ -9,127 +9,42 @@ import (
 	"github.com/quickfeed/quickfeed/scm"
 )
 
-const (
-	private = true
-	public  = !private
-)
-
-// RepoPaths maps from QuickFeed repository path names to a boolean indicating
-// whether or not the repository should be create as public or private.
-var RepoPaths = map[string]bool{
-	qf.InfoRepo:        public,
-	qf.AssignmentsRepo: private,
-	qf.TestsRepo:       private,
-}
-
 // createCourse creates a new course for the directory specified in the request
 // and creates the repositories for the course. Requires that the directory
 // does not contain the QuickFeed repositories that will be created.
 func (s *QuickFeedService) createCourse(ctx context.Context, sc scm.SCM, request *qf.Course) (*qf.Course, error) {
-	org, err := sc.GetOrganization(ctx, &scm.GetOrgOptions{ID: request.OrganizationID})
+	courseCreator, err := s.db.GetUser(request.GetCourseCreatorID())
```

Comment:
Should we instead get the userID of the course creator from context instead of the request?

Reply:
We do get it from context and adding to the request before calling this method, and it still has to be added to the request as well because the request will be used as a query when creating a database record for the new course.

Reply:
Ok. Got it.